### PR TITLE
Update dependencies for Android 34

### PR DESCRIPTION
**Pull request by abhay**

### CODE REVIEW

**Comment by abhay** [comment]
this fails to build locally but appears to build on CI. Needs another look by @bryzettler or @ChewingGlass.

File: tsconfig.json [hunk 1]
```diff
@@ -8,7 +8,6 @@
     "noEmit": true,
     "isolatedModules": true,
     "strict": true,
-    "moduleResolution": "node",
```
Comment (Perronef5) [inline tsconfig.json:11]: @abhay Why did you remove this?

Comment (ChewingGlass) [reply tsconfig.json:11]: because you have to remove it to make it work with the newest tsconfig from react-native

File: android/app/src/main/java/com/helium/wallet/app/MainApplication.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import com.facebook.react.ReactPackage
+import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.load
+import com.facebook.react.defaults.DefaultReactHost.getDefaultReactHost
+import com.facebook.react.defaults.DefaultReactNativeHost
+import com.facebook.soloader.SoLoader
+
+class MainApplication : Application(), ReactApplication {
+
+  override val reactNativeHost: ReactNativeHost =
+      object : DefaultReactNativeHost(this) {
```
Comment (tyler-whitman) [inline android/app/src/main/java/com/helium/wallet/app/MainApplication.kt:17]: I think this needs to use the expo host wrapper since we use their libs:

```
    override val reactNativeHost: ReactNativeHost =
        ReactNativeHostWrapper(this, object : DefaultReactNativeHost(this) {
            override fun getPackages(): List<ReactPackage> {
                val packages: MutableList<ReactPackage> = PackageList(this).packages
                return packages
            }

            override fun getJSMainModuleName(): String = "index"

            override fun getUseDeveloperSupport(): Boolean = BuildConfig.DEBUG

            override val isNewArchEnabled: Boolean = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
            override val isHermesEnabled: Boolean = BuildConfig.IS_HERMES_ENABLED
        })
```

File: android/build.gradle [hunk 3]
```diff
@@ -2,21 +2,21 @@
...
     ext {
-        buildToolsVersion = "33.0.0"
-        minSdkVersion = 24
-        compileSdkVersion = 33
-        targetSdkVersion = 33
-        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
-        ndkVersion = "23.1.7779620"
-        kotlinVersion = "1.6.0"
+        buildToolsVersion = "34.0.0"
+        minSdkVersion = 23
```
Comment (tyler-whitman) [inline android/build.gradle:6]: Lets keep min sdk 24 thats what it was at before. We could even bump it up more if we wanted? Helium Mobile is at 30

File: package.json [hunk 4]
```diff
@@ -13,14 +13,18 @@
     "android-clean": "cd android && ./gradlew clean && cd ..",
     "ios-clean": "rm -rf ios/Pods && rm -rf ios/build",
     "pod-install": "cd ios && pod install && cd ..",
-    "clean-install": "watchman watch-del-all && rm -rf $TMPDIR/react-native-packager-cache-* && rm -rf $TMPDIR/metro-bundler-cache-* && rm -rf node_modules/ && yarn cache clean && yarn install && yarn android-clean && yarn ios-clean && yarn pod-install",
+    "clean-install": "watchman watch-del-all && rm -rf $TMPDIR/react-native-packager-cache-* && [ -d $TMPDIR/metro-bundler-cache-* ] && rm -rf $TMPDIR/metro-bundler-cache-* || true && rm -rf node_modules/ && yarn cache clean && yarn install && yarn android-clean && yarn ios-clean && yarn pod-install",
```
Comment (tyler-whitman) [inline package.json:16]: Found this cool repo https://github.com/pmadruga/react-native-clean-project

can change the clean command to this and it becomes interactive
```
   "clean": "react-native-clean-project",
```

ok to leave the old one if you want too, but this clean project works pretty well.

File: tsconfig.json (hunk 1 above)
Comment (tyler-whitman) [reply tsconfig.json:11]: Yeah we can just follow what we extend `@react-native/typescript-config/tsconfig.json` which sets it to `"moduleResolution": "bundler"`

**Review by ChewingGlass — APPROVED** [review]
